app/src/main/java/org/theotech/ceaselessandroid/person/PersonManagerImpl.java

```diff
@@ -358,19 +366,84 @@ public void populateContacts() {
                         person.setActive(true);
                         ++updated;
                     }
+                    people.add(person);
                 }
             }
         } finally {
             if (cursor != null) {
                 cursor.close();
             }
         }
+
+        // Clean up contacts whose ids have now changed
+        // Copy their contents to an existing contact if there is a matching one.
+        RealmResults<Person> fullList = realm.where(Person.class)
+                .equalTo(Person.Column.ACTIVE, true)
+                .findAll();
+        for (int i = 0; i < fullList.size(); i++) {
+            Person p = fullList.get(i);
+            if (!ids.contains(p.getId())) {
+                Log.v(TAG, "Contact with id " + p.getId() + " no longer exists.");
+                Person matchingContact = findPersonWithName(p.getName(), people);
+                if (matchingContact != null) {
+                    Log.v(TAG, "A contact with name " + p.getName() + " exists. Merging details.");
+                    copyContact(p, matchingContact);
+                    // cleanup the obsolete contact
+                    p.removeFromRealm();
+                } else {
+                    Log.v(TAG, "Marking this contact inactive because it no longer exists on the phone.");
+                    p.setActive(false);
+                }
+            }
+        }
+
         realm.commitTransaction();
 
         Log.d(TAG, String.format("Successfully added %d and updated %d contacts.", added, updated));
         sampleAndPostMetrics();
     }
 
+    private Person findPersonWithName(String name, List<Person> people) {
+        for (Person p : people) {
+            if (p.getName().equals(name)) {
```

Comment:
We aren't checking the actual number, just whether or not it hasPhoneNumber
https://developer.android.com/reference/android/provider/ContactsContract.ContactsColumns.html

The other complexity is that there are many types of phone numbers so we would be recreating the phone number schema in order to detect if two contacts are the same...